### PR TITLE
[Port But Not Really] Rodent Kings from TG

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

Adds rodent kings, an even which spawns rats and hostile rodent kings in maintenance. Their whole purpose is to make the station filthy and to annoy the janitor/service department.

Rat Kings have these following abilities:
* Riot: Spawns a small hostile rat
* Domain: Spawns a little filth, and spreads miasma
* Regeneration: Rats can regenerate health by eating meat, and the regal ones regenerate health if the miasma is high enough
* Rummage: The regal rats can rummage through garbage compactors and pull out trash, or the precious cheese
* Lick: The regal rats can lick glass containers to add a bit of rat spit in them, which makes people disgusted and sick after drunk
* Ventcrawling and rat allegiances: your rats will defend you from intruders, including other regal rats.

Based off https://github.com/tgstation/tgstation/pull/49884

## Why It's Good For The Game

It's fun on TG, and it ways for people to annoy janitors/service people more. Get lynched because your food tastes like feces and it makes people puke!

## Changelog
:cl:
add: regal rats event
/:cl:

### CODE REVIEW

**Comment by zeskorion** [comment]
Hmmmmm... what about adding the same disease spawning code mice have, giving viro incentive to breed rats for disease,  and ability to work without cargo

Edit: wait nevermind. This is an event antag and not like fulp's rodent breeding

**Comment by francinum** [comment]
Players hardly care about keeping the station clean as is, all this is going to do is make that worse but give them a reason to stab rats.

**Comment by badideaguy** [comment]
The regal rats should freak out when see cat people, aka deal 2x more damage to cat people cause cat people ew funny

**Comment by Fronsis** [comment]
Based???' On my Ideas PR i suggested two things:

1- Enable mice breeding again to enable fun gimmicks

2- Get the Rodent King model but make it into a mid-round antag that's like a better version of Nightmare, you lurk around on maint to eat wires and annoy the crew

Honestly i thought he was only a simplemob but if this a mid-round with all those powers.. looks pretty based i would add something to annoy power but not that much

How are his stats? I wouldn't like to get him, get spotted and die by a single laser

**Comment by eeSPee** [comment]
> How are his stats? I wouldn't like to get him, get spotted and die by a single laser

I went with 125 hp and 10 damage. He gets to regen in miasma.

**Comment by PowerfulBacon** [comment]
but they're mice not rats 😳

**Comment by Isy232** [comment]
Give the rodent kings a buff based on how much trash is around, maybe it will force the station to be clean.

**Comment by Unidentified-Biohazard** [comment]
step 1. add cool rat factions

step 2. ????

step 3. Add Skaven as a antag/race/ruin.

**Comment by francinum** [comment]
Add a control config flag and disable it on sage.

**Comment by eeSPee** [comment]
> Add a control config flag and disable it on sage.

Why?

File: code/modules/mob/living/simple_animal/hostile/rats.dm [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+
+/mob/living/simple_animal/hostile/rat/simp/CanAttack(atom/the_target)
+	if(istype(the_target,/mob/living/simple_animal))
+		var/mob/living/A = the_target
+		if(istype(the_target, /mob/living/simple_animal/hostile/rat/king) && A.stat == CONSCIOUS)
+			var/mob/living/simple_animal/hostile/rat/king/ratking = the_target
+			if(ratking.faction_check_mob(src, TRUE))
+				return FALSE
+			else
+				return TRUE
```
Comment (utilbird) [inline code/modules/mob/living/simple_animal/hostile/rats.dm:84]:
```suggestion
			return !ratking.faction_check_mob(src, TRUE)
```

File: code/modules/mob/living/simple_animal/hostile/rats.dm [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+		if(istype(the_target, /mob/living/simple_animal/hostile/rat) && A.stat == CONSCIOUS)
+			var/mob/living/simple_animal/hostile/rat/R = the_target
+			if(R.faction_check_mob(src, TRUE))
+				return FALSE
+			else
+				return TRUE
```
Comment (utilbird) [inline code/modules/mob/living/simple_animal/hostile/rats.dm:90]:
```suggestion
			return !R.faction_check_mob(src, TRUE)
```

File: code/modules/mob/living/simple_animal/hostile/rats.dm [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+	if(istype(the_target,/mob/living/simple_animal))
+		var/mob/living/A = the_target
+		if(istype(the_target, /mob/living/simple_animal/hostile/rat/king) && A.stat == CONSCIOUS)
+			return TRUE
+		if(istype(the_target, /mob/living/simple_animal/hostile/rat) && A.stat == CONSCIOUS)
+			var/mob/living/simple_animal/hostile/rat/R = the_target
+			if(R.faction_check_mob(src, TRUE))
+				return FALSE
+			else
+				return TRUE
```
Comment (utilbird) [inline code/modules/mob/living/simple_animal/hostile/rats.dm:173]:
```suggestion
			return !R.faction_check_mob(src, TRUE)
```

File: code/modules/mob/living/simple_animal/hostile/rats.dm [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+
+	if (target.reagents && istype(target,/obj) && target.is_injectable(src,TRUE))
+		src.visible_message("<span class='warning'>[src] starts licking the [target] passionately!</span>","<span class='notice'>You start licking the [target]...</span>")
+		rummaging = TRUE
+		if (do_after(src,2 SECONDS, target) && target)
+			target.reagents.add_reagent(/datum/reagent/rat_spit,1,no_react = TRUE)
+			to_chat(src, "<span class='notice'>You finish licking the [target].</span>")
+		rummaging = FALSE
+		return
+	. = ..()
```
Comment (utilbird) [inline code/modules/mob/living/simple_animal/hostile/rats.dm:225]:
```suggestion
	return ..()
```

**Comment by eeSPee** [comment]
Well the changes to regalrats got ported to tg now, so this makes it a complete port from tg?

**Comment by TeomanTheGreat** [comment]
This should be reopened.

**Pull request closed without merging**